main.go

```diff
@@ -202,9 +210,30 @@ func LoadBundles (wh warehouse.Warehouse, filename string, bundles ...fullstory.
 	return nil
 }
 
+func getExportData(fs *fullstory.Client, bundleID int) (fullstory.ExportData, error) {
+	log.Printf("Getting Export Data for bundle %d\n", bundleID)
+	for r := 1; r <= maxAttempts; r++ {
+		stream, err := fs.ExportData(bundleID)
+		if err != nil {
+			log.Printf("Failed to fetch export data for Bundle %d", bundleID)
+			retryAfterDuration := defaultRetryAfterDuration
+			if statusError, ok := err.(fullstory.StatusError); ok {
+				if retryAfter, err := strconv.Atoi(statusError.RetryAfter); err == nil {
+					retryAfterDuration = time.Duration(retryAfter) * time.Second
+				}
+			}
+			log.Printf("Attempt #%d failed. Retrying after %s\n", r, retryAfterDuration)
+			time.Sleep(retryAfterDuration)
```

Comment:
Hauser may receive errors with non-5XX status codes. In which case, retrying will not help. 5XX error codes are returned when something went wrong on the server side and the request can be tried again in some time.

So, I guess this case, we can check if a `Retry-After` header exists and wait for the specified time. If there was no such header, check if it was a `5XX` error and retry after the default time. Exit early for all other errors.